crates/nu-parser/src/lite_parse.rs

```diff
@@ -65,12 +65,20 @@ fn bare(src: &mut Input, span_offset: usize) -> Result<Spanned<String>, ParseErr
         0
     };
 
+    let mut escaping = false;
     let mut inside_quote: Option<char> = None;
     let mut block_level: Vec<BlockKind> = vec![];
 
     while let Some((_, c)) = src.peek() {
         let c = *c;
-        if inside_quote.is_some() {
+
+        if escaping {
+            escaping = false;
```

Comment:
This would mean, for example, that `\t` would be a `t`, right? That would be unfamiliar with "backslash style" escaping.

Reply:
Yes, \t is a t in other shells including inside backticks, you can test by doing e.g. `ls `p\wd` in bash for example.

From what I gather, `\<anything>` isn't considered an escape only if it's inside double or single quotes "traditionally".

As for completing `te'st` without adding the escape, that would not really work, because you then end up with a parse error, missing ending `'`

Reply:
What I meant was that `\t` is often a tab, `\n` a newline, and so on:

```
❯ echo "\tfoo"
	foo
```

Note that is a tab character plus "foo", not "tfoo". 

> From what I gather, \<anything> isn't considered an escape only if it's inside double or single quotes "traditionally".

That isn't the case, actually. You can write this `filename\ with\ spaces.txt`, for example, without being in quotes. Mind you, if we decided we wanted to implement this style of escaping, we can build up the implementation (in other words, we don't have to cover all possible cases in one PR 🙂 )

> As for completing `te'st` without adding the escape, that would not really work, because you then end up with a parse error, missing ending `'`

I'm confused. What would the parse error be? `"test'/"` and `test\'/` should be equivalent strings.

Reply:
Ah, `\t` is a `t` when it's not inside quotes though.

I find all of these quoting and escaping rules more confusing than helpful. You can get used to them, but they don't feel expressive or intuitive. I think that's why we've been holding off on doing backslash escaping.

Reply:
> What I meant was that `\t` is often a tab, `\n` a newline, and so on:
> 
> ```
> ❯ echo "\tfoo"
> 	foo
> ```
> 
> Note that is a tab character plus "foo", not "tfoo".

Only for "gnulike" programs, not for shells. Echo "1\t2" prints `1\t2` only because echo understands the string `\t` as `<TAB>` the shell doesn't do anything here because it's in quotes (as per my previous comment).
You can test this by just doing say `foo\t` with no echo and you'll get `foot command not found`

> 
> > From what I gather, <anything> isn't considered an escape only if it's inside double or single quotes "traditionally".
> 
> That isn't the case, actually. You can write this `filename\ with\ spaces.txt`, for example, without being in quotes. Mind you, if we decided we wanted to implement this style of escaping, we can build up the implementation (in other words, we don't have to cover all possible cases in one PR  )

You got it the other way around. Escapes are only applied in shells when not inside quotes or when in backtick quotes. Double and single quoted values will not escpape due to `\` (I'm not 100% sure about windows on this).

> 
> > As for completing `te'st` without adding the escape, that would not really work, because you then end up with a parse error, missing ending `'`
> 
> I'm confused. What would the parse error be? `"test'/"` and `test\'/` should be equivalent strings.

You'd get a parser error saying it expected a terminating `'` AFAICS. Again you can test this in other shells too, try doing `ls /tmp/te'st` (provided it exists) without the escape char.

Reply:
Yeah, my bad, I edited my comments because I misunderstood the "bare word" context.

> You'd get a parser error saying it expected a terminating ' AFAICS. Again you can test this in other shells too, try doing `ls /tmp/te'st` (provided it exists) without the escape char.

To make it comparable to what I'm suggesting, try doing `ls "/tmp/te'st"`. That should work, no?

Reply:
> Yeah, my bad, I edited my comments because I misunderstood the "bare word" context.
> 
> > You'd get a parser error saying it expected a terminating ' AFAICS. Again you can test this in other shells too, try doing `ls /tmp/te'st` (provided it exists) without the escape char.
> 
> To make it comparable to what I'm suggesting, try doing `ls "/tmp/te'st"`. That should work, no?

Yes that should work, and it works with this change too.